tensorpipe/test/channel/channel_test.cc

```diff
@@ -333,92 +327,3 @@ TEST_P(ChannelTest, contextIsNotJoined) {
         context->createChannel(std::move(conn), Channel::Endpoint::kConnect);
       });
 }
-
-TEST_P(ChannelTest, CallbacksAreDeferred) {
```

Comment:
This was deleted because it is testing implementation rather than behavior. We can bring it back, but then we'll need some way to exclude it for some channels (CUDA-based for instance, as copy happening asynchronously means the correlation between "callback not called inline" and "send doesn't do the heavy lifting" does not hold).

Reply:
Agree that this test is an imperfect proxy for what we really mean to test, but it is still a "valid" test: it is performing a legal series of operations whose semantics is well defined and whose behavior can thus be tested. Right? Or am I misremembering what this test does?

If you have a way in mind to test the intended behavior then we can add such a test and replace this, but I'd rather not remove it without a replacement.